pkg/credentialprovider/config.go

```diff
@@ -122,29 +122,38 @@ func ReadDockerConfigJSONFile(searchPaths []string) (cfg DockerConfig, err error
 	if len(searchPaths) == 0 {
 		searchPaths = DefaultDockerConfigJSONPaths()
 	}
-
 	for _, configPath := range searchPaths {
 		absDockerConfigFileLocation, err := filepath.Abs(filepath.Join(configPath, configJsonFileName))
 		if err != nil {
 			glog.Errorf("while trying to canonicalize %s: %v", configPath, err)
 			continue
 		}
-		glog.V(4).Infof("looking for .docker/config.json at %s", absDockerConfigFileLocation)
-		contents, err := ioutil.ReadFile(absDockerConfigFileLocation)
-		if os.IsNotExist(err) {
-			continue
-		}
-		if err != nil {
-			glog.V(4).Infof("while trying to read %s: %v", absDockerConfigFileLocation, err)
-			continue
-		}
-		cfg, err := readDockerConfigJsonFileFromBytes(contents)
-		if err == nil {
-			glog.V(4).Infof("found .docker/config.json at %s", absDockerConfigFileLocation)
+		glog.V(4).Infof("looking for %s at %s", configJsonFileName, absDockerConfigFileLocation)
+		if cfg, err = ReadSpecificDockerConfigJsonFile(absDockerConfigFileLocation); err == nil {
 			return cfg, nil
 		}
 	}
-	return nil, fmt.Errorf("couldn't find valid .docker/config.json after checking in %v", searchPaths)
+	return nil, fmt.Errorf("couldn't find valid %s after checking in %v", configJsonFileName, searchPaths)
+
+}
+
+//ReadDockerConfigFromJsonFormat attempts to read docker configuration from a given json fromat file.
+func ReadSpecificDockerConfigJsonFile(filePath string) (cfg DockerConfig, err error) {
+	var contents []byte
+	contents, err = ioutil.ReadFile(filePath)
+	if os.IsNotExist(err) {
+		return nil, err
+	}
+	if err != nil {
+		glog.V(4).Infof("while trying to read %s: %v", filePath, err)
```

Comment:
don't log in the utility method, just return the error and do the `os.IsNotExist` check and logging in the loop above